Changed rebuild order and added rebuild indicator

Alaschinger opened the pull request:
The rebuild order has been changed, so that opencast can be used while the workflows are still updating. Still active workflows won't be updated during the rebuild, because they will automatically be updated when finished.

Since not all information in the admin UI is displayed correctly until the workflow rebuild has finished an index rebuild indicator has been added to the admin UI frontend. This indicator is getting the states of the repopulation services from the IndexRebuildService using a new endpoint. They are shown to the user within the header of the admin UI. If no rebuild is running the indicator is hidden.

Alaschinger committed:
index rebuild indicator and changed rebuild order

The rebuild order has been changed, so that opencast can be used while the workflows are still updated. Since not all information in the admin UI is displayed correctly until the workflow rebuild has finished an index rebuild indicator has been added to the admin UI frontend. This indicator is getting the states of the repopulation service from the IndexRebuildService using a new endpoint and shows them to the user within the header of the admin UI. If no rebuild is running the indicator is hidden.

gregorydlogan commented:
Looks like GH missed this conflict.

mtneug commented:
Since this PR requires changes to `etc/security/mh_default_org.xml`, it can probably no longer go into 13.0.

gregorydlogan commented:
Alternatively, as long as the counter doesn't break the ui if it can't get at the endpoint it still could...

Arnei commented:
Since Greg suggested this might still make it into 13.0 if the index rebuild indicator does not break the UI if the endpoint is unreachable (it does not), I took the liberty of rebasing this onto r/13.x while fixing the merge conflict. @mtneug what do you think?

mtneug commented:
If there is no error when running with an old `etc/security/mh_default_org.xml` than this is fine for me 👍

gregorydlogan commented on modules/admin-ui-frontend/.jshintrc:
```diff
@@ -1,16 +0,0 @@
-{
```
This file is required by the current state of `r/13.x`

gregorydlogan replied:
To be clear:
```
[INFO] 16 01 2023 12:33:13.247:INFO [framework.detect-browsers]: The following browsers will be used: [ 'ChromeHeadless', 'FirefoxHeadless' ]
[INFO] 16 01 2023 12:33:13.437:WARN [filelist]: Pattern "/home/greg/opencast/upstream/modules/admin-ui-frontend/test/test/unit/**/*Helper.js" does not match any file.
```

Arnei replied:
Added it back, tests should run again now.

gregorydlogan commented on modules/admin-ui-frontend/app/scripts/shared/services/indexRebuild.js:
```diff
@@ -0,0 +1,115 @@
+          rebuildServices.showRebuildStates = true;
+        }
+      });
+    }).catch(function (err) {
```
This exception will cause the UI element to incorrectly indicated that there's an issue if an old security xml is in use.  Two choices here:

1) Parse out the error, if it's a 503 then it's *probably* just old security and now you get to figure out how to tell the user that.
2) Move this PR into `develop`

Arnei replied:
Well then, rebasing on `develop` it is.

Arnei committed:
Add .jshintrc back

The file is apparently still used somewhere in the depths of the admin
ui and its abscence will cause the tests to fail.
